File: composeApp/src/androidMain/kotlin/com/muralex/exploramus/screens/favorites/FavoritesScreen.kt
package com.muralex.exploramus.screens.favorites

import androidx.compose.runtime.Composable
import com.muralex.exploramus.screens.favorites.views.FavoritesGrid
import com.muralex.exploramus.screens.favorites.views.FavoritesList
import com.muralex.exploramus.ui.adaptive.LocalFormFactor
import com.muralex.exploramus.ui.adaptive.isCompact
import com.muralex.exploramus.ui.components.EmptyState
import com.muralex.exploramus.ui.components.EmptyStateView
import com.muralex.exploramus.ui.components.FadeLoadingContent
import com.muralex.exploramus.viewmodel.screens.favorites.FavoritesScreenState


@Composable
fun FavoritesScreen(
    screenState: FavoritesScreenState,
    eventHandler: FavoritesEventHandler
) {
    FadeLoadingContent(isLoading = screenState.isLoading) {
        FavoritesScreenContent(
            screenState = screenState,
            onEvent = eventHandler::onEvent,
        )
    }
}

@Composable
fun FavoritesScreenContent(
    screenState: FavoritesScreenState,
    onEvent: (FavoritesUiEvent) -> Unit,
) {
    val formFactor = LocalFormFactor.current
    val useGrid = !formFactor.isCompact

    if (screenState.favorites.isEmpty()) {
        EmptyStateView(EmptyState.EmptyList)
    } else {
        if (useGrid) {
            FavoritesGrid(
                gridItems = screenState.favorites,
                onEvent = onEvent
            )
        } else {
            FavoritesList(
                listItems = screenState.favorites,
                onEvent = onEvent
            )
        }
    }
}

File: iosApp/iosApp/views/screens/favorites/FavoritesScreen.swift

import SwiftUI
import Shared

struct FavoritesScreen: View {
    let screenState: FavoritesScreenState
    let eventHandler: FavoritesEventHandler
    @Environment(\.appLayout) var appLayout

    var body: some View {
        Level1ScreenContainer (screenTitle: Strings.favoritesTitle) {
            ZStack {
                Color.clear
                if screenState.isLoading {
                    LoadingScreen()
                        .transition(.opacity)
                } else {
                    FavoritesScreenContent(
                        screenState: screenState,
                        onEvent: eventHandler.onEvent
                    )
                    .transition(.opacity)
                }
            }
            .animation(.easeInOut(duration: 0.15), value: screenState.isLoading)
            .toolbar {
                ToolbarItem(placement: .principal) {
                    Text(appLayout.isTablet ? "" : Strings.favoritesTitle)
                        .font(.title2)
                        .fontWeight(.semibold)
                }
            }
            .navigationBarTitleDisplayMode(.inline)
        }
        
    }
}


struct FavoritesScreenContent: View {
    let screenState: FavoritesScreenState
    let onEvent: (FavoritesUiEvent) -> Void
    @Environment(\.appLayout) var appLayout

    var body: some View {
        ZStack {
            if screenState.favorites.isEmpty {
                EmptyStateView(state: .emptyList)
                    .transition(.opacity)
            } else if appLayout.isTablet {
                FavoritesGrid(
                    items: screenState.favorites,
                    onEvent: onEvent
                )
                .transition(.opacity)
            } else {
                FavoritesList(
                    items: screenState.favorites,
                    onEvent: onEvent
                )
                .transition(.opacity)
            }
        }
        .animation(.easeInOut(duration: 0.15), value: screenState.favorites.isEmpty)
        
    }
}









 

 

